### ISSUE TITLE
Run SKILL tests using Pytest

### ISSUE THREAD

**Issue by curtisma** [cascode-labs/virtue#37]
**Problem Statement**
----------------------------

It would be great to be able to run the SKILL tests by mapping each SKILL test function to a pytest test case.  This would allow the results to show up in VS Code's test runner for example.

**Proposed Solution**
-------------------------------------------

The user should be able to write a single python script which contains a test function which generates a test case for each SKILL test function.  The user should be able to specify files and or directories to be included.

https://stackoverflow.com/questions/4923836/generating-py-test-tests-in-python
https://docs.pytest.org/en/latest/how-to/parametrize.html#pytest-generate-tests

**Alternatives Considered**
-----------------------------------------------

- Write the tests in pure python and just make the function calls in SKILL through SKILL bridge.  However this makes it harder to debug.  The SKILL debugger built-in to Virtuoso is still the best option I've found for debugging SKILL code.

**Proposed Action Plan**
-------------------------------
- [ ] 1. we'll do this
- [ ] 2. We'll do that
- [ ] 3. Then we'll add tests for this and that.

**Comment by tbrandtner** [comment]
I agree that the Skill debugger is the best way for debugging. But running automatic tests and show their results in an already existing environment (vscode or Jenkins) would be also quite helpful. 
What about a solution where you define your tests still in Skill in a "standardized" way - but adding an easy way to call each of these Skill tests also from Python which could track the results for vscode, Jenkins etc.
I suppose that we need to call the single Skill test procedures via skillbridge because we cannot afford to launch Virtuoso again and again.
